plugin/evm/uptime/pausable_manager_test.go

```diff
@@ -0,0 +1,205 @@
+		checkUptime(t, up, nodeID0, 0*time.Second, currentTime)
+
+		// Disconnect and check uptime
+		currentTime = addTime(clk, 3*time.Second)
+		require.NoError(up.Disconnect(nodeID0))
+		// Uptime should not have increased since the node was paused
+		checkUptime(t, up, nodeID0, 0*time.Second, currentTime)
+
+		// Connect again and resume
+		addTime(clk, 4*time.Second)
+		require.NoError(up.Connect(nodeID0))
+		addTime(clk, 5*time.Second)
+		up.OnValidatorStatusUpdated(vID, nodeID0, true)
+
+		// Check uptime after resume
+		currentTime = addTime(clk, 6*time.Second)
+		// Uptime should have increased by 6 seconds since the node was resumed
+		checkUptime(t, up, nodeID0, 6*time.Second, currentTime)
+	}
+
+	// Case 4: Start tracking, connect, pause, stop tracking, resume tracking.
+	{
+		up, clk, s := setupTestEnv(nodeID0, startTime)
+
+		// Start tracking and connect
+		require.NoError(up.StartTracking([]ids.NodeID{nodeID0}))
+		addTime(clk, time.Second)
+		require.NoError(up.Connect(nodeID0))
+
+		// Pause and check uptime
+		currentTime := addTime(clk, 2*time.Second)
+		up.OnValidatorStatusUpdated(vID, nodeID0, false)
+		// Uptime should be 2 seconds since the node was paused after 2 seconds
+		checkUptime(t, up, nodeID0, 2*time.Second, currentTime)
+
+		// Stop tracking and reinitialize manager
+		currentTime = addTime(clk, 3*time.Second)
+		require.NoError(up.StopTracking([]ids.NodeID{nodeID0}))
+		up = NewPausableManager(uptime.NewManager(s, clk))
+
+		// Uptime should not have increased since the node was paused
+		// and we have not started tracking again
+		checkUptime(t, up, nodeID0, 2*time.Second, currentTime)
+
+		// Pause and check uptime
+		up.OnValidatorStatusUpdated(vID, nodeID0, false)
+		// Uptime should not have increased since the node was paused
+		checkUptime(t, up, nodeID0, 2*time.Second, currentTime)
+
+		// Resume and check uptime
+		currentTime = addTime(clk, 5*time.Second)
+		up.OnValidatorStatusUpdated(vID, nodeID0, true)
+		// Uptime should have increased by 5 seconds since the node was resumed
+		checkUptime(t, up, nodeID0, 7*time.Second, currentTime)
+
+		// Start tracking and check elapsed time
+		currentTime = addTime(clk, 6*time.Second)
+		require.NoError(up.StartTracking([]ids.NodeID{nodeID0}))
+		// Uptime should have increased by 6 seconds since we started tracking
+		// and node was resumed (we assume the node was online until we started tracking)
+		checkUptime(t, up, nodeID0, 13*time.Second, currentTime)
+
+		// Elapsed time
+		currentTime = addTime(clk, 7*time.Second)
+		// Uptime should not have increased since the node was not connected
+		checkUptime(t, up, nodeID0, 13*time.Second, currentTime)
+
+		// Connect and final uptime check
+		require.NoError(up.Connect(nodeID0))
+		currentTime = addTime(clk, 8*time.Second)
+		// Uptime should have increased by 8 seconds since the node was connected
+		checkUptime(t, up, nodeID0, 21*time.Second, currentTime)
```

Comment:
(optional) imo this test would read slightly better if we had a running total for `expectedUptime` then each time calling `checkUptime(t, up, nodeID0, expectedUptime, currentTime)`

Reply:
changed it